### PR TITLE
Test: domain test maintenance

### PR DESCRIPTION
**Pull request by prtandrup**
## Description

This PR:
* Structures the domain tests in `ChargeIngestionTestChargeInformation` and `ChargeIngestionTestChargePrices` to provide a better overview
* Adds a single rejection test to Charge Information
* Adds a single rejection test to Charge Prices (rejection upon business validation (irregular price series))
* Removes duplicate/overlapping tests
* Improves test and class names for better readability
* Removes unused test files

## References

* #1735

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTest.Core/TestFiles/Charges/ChargeInformationRequest.cs [hunk 1]
```diff
@@ -14,29 +14,25 @@
 
 namespace GreenEnergyHub.Charges.IntegrationTest.Core.TestFiles.Charges
 {
-    public static class ChargeDocument
+    public static class ChargeInformationRequest
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTest.Core/TestFiles/Charges/ChargeInformationRequest.cs:17]: ChargeInformationRequests? or ChargeInformationRequestDocuments?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTest.Core/TestFiles/Charges/ChargeInformationRequest.cs [hunk 2]
```diff
@@ -14,29 +14,25 @@
...
-        public const string TariffSystemOperatorCreate = "TestFiles/Charges/TariffSystemOperatorCreate.xml";
+        public const string BundleWithMultipleOperationsForSameTariff = "TestFiles/Charges/BundleWithMultipleOperationsForSameTariff.xml";
+        public const string BundleWithTwoOperationsForSameTariffSecondOpViolatingVr903 = "TestFiles/Charges/BundleWithTwoOperationsForSameTariffSecondOpViolatingVr903.xml";
 
         public const string ChargeInformationTariffHourlySample = "TestFiles/Samples/Charges/ChargeInformationTariffHourlySample.xml";
         public const string ChargeInformationFeeMonthlySample = "TestFiles/Samples/Charges/ChargeInformationFeeMonthlySample.xml";
         public const string ChargeInformationSubscriptionMonthlySample = "TestFiles/Samples/Charges/ChargeInformationSubscriptionMonthlySample.xml";
         public const string BundledChargeInformationSample = "TestFiles/Samples/Charges/BundledChargeInformationSample.xml";
+
+        // Charge prices below - pending to be moved to new ChargePricesRequest and general clean-up
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTest.Core/TestFiles/Charges/ChargeInformationRequest.cs:34]: Are you planning to do the split in this PR?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTest.Core/TestFiles/Charges/InvalidTaxTariffAsGridAccessProvider.xml [hunk 3]
```diff
@@ -30,14 +30,14 @@ limitations under the License.
             <cim:ChargeType>
                 <cim:chargeTypeOwner_MarketParticipant.mRID codingScheme="A10">8100000000030</cim:chargeTypeOwner_MarketParticipant.mRID>
                 <cim:type>D03</cim:type>
-                <cim:mRID>ChargeId01</cim:mRID>
-                <cim:name>Charge tariff day Name1</cim:name>
-                <cim:description>UPDATE</cim:description>
+                <cim:mRID>{{$randomCharacters}}</cim:mRID>
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTest.Core/TestFiles/Charges/InvalidTaxTariffAsGridAccessProvider.xml:33]: Good change - will lower the risk of different tests influencing each other.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTest.Core/TestFiles/Charges/InvalidTaxTariffAsGridAccessProvider.xml [hunk 4]
```diff
@@ -30,14 +30,14 @@ limitations under the License.
...
+                <cim:name>Invalid tariff</cim:name>
+                <cim:description>Grid Access Provider not allowed to create tax tariff - VR508</cim:description>
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTest.Core/TestFiles/Charges/InvalidTaxTariffAsGridAccessProvider.xml:35]: Nice description of the expected validation error

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTest.Core/TestFiles/Charges/InvalidTaxTariffAsGridAccessProvider.xml [hunk 5]
```diff
@@ -30,14 +30,14 @@ limitations under the License.
...
                 <cim:priceTimeFrame_Period.resolution>P1D</cim:priceTimeFrame_Period.resolution>
-                <cim:effectiveDate>{{$isoTimestampPlusOneMonthAndTwoDays}}</cim:effectiveDate>
-                <cim:VATPayer>D02</cim:VATPayer>
-                <cim:transparentInvoicing>false</cim:transparentInvoicing>
-                <cim:taxIndicator>false</cim:taxIndicator>
+                <cim:effectiveDate>2023-01-14T23:00:00Z</cim:effectiveDate>
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTest.Core/TestFiles/Charges/InvalidTaxTariffAsGridAccessProvider.xml:37]: Personally, I'm not a fan of hard coded dates/times in test-data.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/ChargeIngestionTestsChargeInformation.cs [hunk 6]
```diff
@@ -115,51 +96,76 @@ public async Task Given_NewMessage_When_SenderIdDoesNotMatchAuthenticatedId_Then
...
+                var (request, correlationId) = Fixture.AsSystemOperator.PrepareHttpPostRequestWithAuthorization(
+                    EndpointUrl, ChargeInformationRequest.TaxTariffAsSystemOperator);
 
                 // Act
-                var actual = await Fixture.HostManager.HttpClient.SendAsync(updateReq);
+                await Fixture.HostManager.HttpClient.SendAsync(request);
 
                 // Assert
                 using var assertionScope = new AssertionScope();
-                actual.StatusCode.Should().Be(HttpStatusCode.Accepted);
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/ChargeIngestionTestsChargeInformation.cs:142]: To me, the `actual.StatusCode` assertion actually holds value, especially when debugging the tests.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/ChargeIngestionTestsChargeInformation.cs [hunk 7]
```diff
@@ -115,51 +96,76 @@ public async Task Given_NewMessage_When_SenderIdDoesNotMatchAuthenticatedId_Then
...
+                // Assert
+                using var assertionScope = new AssertionScope();
                 var peekResults =
-                    await Fixture.MessageHubMock.AssertPeekReceivesRepliesAsync(
-                        updateCorrelationId,
-                        ResponseFormat.Xml);
-                peekResults.Should().ContainMatch("*ConfirmRequestChangeOfPriceList_MarketDocument*");
+                    await Fixture.MessageHubMock.AssertPeekReceivesRepliesAsync(correlationId, ResponseFormat.Json, 8);
+
+                peekResults.Should().NotContainMatch("*RejectRequestChangeOfPriceList_MarketDocument*");
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/ChargeIngestionTestsChargeInformation.cs:146]: Alternatively, we could deserialize the JSON and use the object for assertions. Not sure if it is better or prettier, just want to make sure you have weighed the possibilities.

**Review by HenrikSommer — COMMENTED** [review]
Comments left for the WIP.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTest.Core/TestFiles/Charges/ChargeInformationRequest.cs (hunk 2 above)
Comment (prtandrup) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTest.Core/TestFiles/Charges/ChargeInformationRequest.cs]: Yup, the split is done [here](https://github.com/Energinet-DataHub/geh-charges/pull/1742/commits/8a518d2bb761d414235c6de6a99173f9b03fc936).

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/ChargeIngestionTestsChargeInformation.cs (hunk 6 above)
Comment (prtandrup) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/ChargeIngestionTestsChargeInformation.cs:142]: Think this was a mistake, added again in https://github.com/Energinet-DataHub/geh-charges/pull/1742/commits/043a52ef92b71f1c6e366440417e7209f44c2423

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/ChargeIngestionTestsChargeInformation.cs (hunk 7 above)
Comment (prtandrup) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/ChargeIngestionTestsChargeInformation.cs:160]: Good point. I'll prefer to do the JSON optimization in a subsequent PR, where I'll add the chronological order test as well.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTest.Core/TestFiles/Charges/PriceSeries/TaxTariffPriceSeries.xml [hunk 8]
```diff
@@ -19,18 +16,18 @@ limitations under the License.
     <cim:type>D10</cim:type>
     <cim:process.processType>D08</cim:process.processType>
     <cim:businessSector.type>23</cim:businessSector.type>
-    <cim:sender_MarketParticipant.mRID codingScheme="A10">{{$senderMarketParticipant}}</cim:sender_MarketParticipant.mRID>
-    <cim:sender_MarketParticipant.marketRole.type>DDM</cim:sender_MarketParticipant.marketRole.type>
+    <cim:sender_MarketParticipant.mRID codingScheme="A10">5790000432752</cim:sender_MarketParticipant.mRID>
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTest.Core/TestFiles/Charges/PriceSeries/TaxTariffPriceSeries.xml:19]:
```suggestion
    <cim:sender_MarketParticipant.mRID codingScheme="A10">{{$systemOperatorMarketParticipant}}</cim:sender_MarketParticipant.mRID>
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTest.Core/TestFiles/Charges/PriceSeries/TaxTariffPriceSeries.xml [hunk 9]
```diff
@@ -19,18 +16,18 @@ limitations under the License.
...
+    <cim:sender_MarketParticipant.marketRole.type>EZ</cim:sender_MarketParticipant.marketRole.type>
     <cim:receiver_MarketParticipant.mRID codingScheme="A10">{{$receiverMarketParticipant}}</cim:receiver_MarketParticipant.mRID>
     <cim:receiver_MarketParticipant.marketRole.type>DDZ</cim:receiver_MarketParticipant.marketRole.type>
     <cim:createdDateTime>{{$isoTimestamp}}</cim:createdDateTime>
     <cim:MktActivityRecord>
         <cim:mRID>OpId1_{{$isoTimestamp}}</cim:mRID>
         <cim:ChargeGroup>
             <cim:ChargeType>
-                <cim:chargeTypeOwner_MarketParticipant.mRID codingScheme="A10">{{$senderMarketParticipant}}</cim:chargeTypeOwner_MarketParticipant.mRID>
+                <cim:chargeTypeOwner_MarketParticipant.mRID codingScheme="A10">5790000432752</cim:chargeTypeOwner_MarketParticipant.mRID>
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTest.Core/TestFiles/Charges/PriceSeries/TaxTariffPriceSeries.xml:28]:
```suggestion
                <cim:chargeTypeOwner_MarketParticipant.mRID codingScheme="A10">{{$systemOperatorMarketParticipant}}</cim:chargeTypeOwner_MarketParticipant.mRID>
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTest.Core/TestFiles/Charges/TariffAsSystemOperator.xml [hunk 10]
```diff
@@ -14,11 +17,11 @@ limitations under the License.
 <cim:RequestChangeOfPriceList_MarketDocument xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:cim="urn:ediel.org:structure:requestchangeofpricelist:0:1" xsi:schemaLocation="urn:ediel.org:structure:requestchangeofpricelist:0:1 urn-ediel-org-structure-requestchangeofpricelist-0-1.xsd">
     <cim:mRID>DocId{{$isoTimestamp}}</cim:mRID>
     <cim:type>D10</cim:type>
-    <cim:process.processType>D08</cim:process.processType>
+    <cim:process.processType>D18</cim:process.processType>
     <cim:businessSector.type>23</cim:businessSector.type>
     <cim:sender_MarketParticipant.mRID codingScheme="A10">5790000432752</cim:sender_MarketParticipant.mRID>
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTest.Core/TestFiles/Charges/TariffAsSystemOperator.xml:22]:
```suggestion
    <cim:sender_MarketParticipant.mRID codingScheme="A10">{{$systemOperatorMarketParticipant}}</cim:sender_MarketParticipant.mRID>
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTest.Core/TestFiles/Charges/TariffAsSystemOperator.xml [hunk 11]
```diff
@@ -27,21 +30,15 @@ limitations under the License.
             <cim:ChargeType>
                 <cim:chargeTypeOwner_MarketParticipant.mRID codingScheme="A10">5790000432752</cim:chargeTypeOwner_MarketParticipant.mRID>
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTest.Core/TestFiles/Charges/TariffAsSystemOperator.xml:31]:
```suggestion
                <cim:chargeTypeOwner_MarketParticipant.mRID codingScheme="A10">{{$systemOperatorMarketParticipant}}</cim:chargeTypeOwner_MarketParticipant.mRID>
```

**Review by HenrikSommer — APPROVED** [review]

**Pull request merged**